src/app/common/confirmation-modal/confirm-dialog.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { ConfirmationModalComponent } from './confirmation-modal.component';
+import { IConfirmationModalData } from '../model/confirmation-modal.types';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConfirmDialogService {
+  public constructor(private readonly dialog: MatDialog) {
+  }
+
+  public openConfirmModal(config: IConfirmationModalData): Observable<boolean | undefined> {
```

Comment:
It should never be `undefined`. If user canceled the modal instead selecting 'Yes' or 'No' and we get `undefined` it means we can return `false` from this method as confirm modal was not confirmed. Note also that even if we have only two buttons and clicking on any of them should result in `true` / `false` user can still cancel modal by using escape button and then this will result in `undefined`.